designs/authentication-use-cases.md

```diff
@@ -0,0 +1,96 @@
+ability to find credentials locally rather than requiring them to be supplied in the allowedCredentials
+set in authentication.
+
+* User Verification - https://www.w3.org/TR/webauthn-2/#user-verification
+
+The process of supplying extra or supplementary authentication to the authenticator. This improves
+the authentication from "there is a person present" to "this specific owner of the authenticator" is
+present. Conseder UV=True as the authenticator is a self container MFA device, where UV=false means that
+the device is a SFA, and other authentication is required to compose a complete MFA.
+
+At registration we store the policy and verification that was supplied in the internal Credential
+type so that later decisions can be made for security policy.
+
+## Authentication Work Flows
+
+### Homogenous Non-Discoverable Credentials
+
+This is a set of credentials with a homogeneous UV policy - either all discouraged or all required.
+
+This policy is derived from the set of UV=true/false flags in the allow credential list. If the
+UV flags are inconsistent an error is raised.
+
+In the case all of these credentials are discouraged, we still assert that the UV bit matches
+the returned authentication as it *is* valid for UV=true even when discouraged and we store that
+in the credential.
+
+* Example
+
+A laptop where the user has multiple UV=false yubikeys that may be used as a authentication factor.
+
+A user indicates they want to authenticate with their touchid as a self contained MFA. The user owns
+and has enrolled multiple devices that have TouchID so any of them could be the credential in use
+without external knowledge.
+
+
+### Single Credential Non-Discoverable with UV Policy Override
+
+If external knowledge is provided about which credential is intended to be used, we can have a credential's
+requirements for verification temporarily overriden. This can be to remove the UV requirement, or
+to temporarily require it.
+
+* Example
+
+Using a yubikey with pin as an MFA device, after login is used for privilege escalation where we only
+need to assert a single factor (presence) at they are already authenticated.
+
+
+### Single Credential Discoverable with UV Policy Override
+
+With device specific information we can use a discoverable credential with policy override if required.
+This could be considered as a device specific credential / authentication mechanism.
+
+The other elements of "Single Credential Non-Discoverable with UV Policy Override" remain.
+
+* Example
+
+A longer term cookie on the device, or the device finger print indicates to the server that we can
+use a discoverable credential from that device.
+
+## Work flows that may not be relevant?
+
+### Homogenous Discoverable Credentials
+
+As a discoverable credential we already required external knowledge of what credential will be used
+so does it become needed to have many discoverable credentials in one transaction?
```

Comment:
possibly. touchid on my iphone + mac. the rp shouldn't have to know which device the user is on but wants to accept both. am i missing something here?

Reply:
Yeah, I thought about that possibility too. From my understanding, the intent is for a discoverable credential to be used, that the server has "pre-knowledge" about the client so it knows "what to expect" in response. This could be a device id, a cookie, fingerprint ...

But I can certainly also see how without the external knowledge, we'd need to accept "well it could be any device now". So that's why it's a "question" if we need this or not? Again, not opposed, but it's good to just have those work flows, and I think we can expand this based on your suggestion, and the fact we make not have pre-knowledge.

Saying this, how would this interact with non-discoverable credentials? How would we have a situation where we say and present a UI that says "okay, what device did you want to use?" I think that may be my concern. We can't mix discoverable and non-discoverable in one transaction, so that's where the "external knoweldge" is needed, and once you have that external knowledge, you may as well select the credential matching that device?